spec/function.dd

```diff
@@ -151,33 +151,48 @@ $(GNAME OutStatement):
     $(D out) $(D $(LPAREN)) $(GLINK_LEX Identifier) $(D $(RPAREN)) $(GLINK2 statement, BlockStatement)
 )
 
-$(H2 $(LNAME2 contracts, Contracts))
+        $(P Function Contracts specify the preconditions and postconditions of a function.
+        They are used in $(LINK2 contracts.html, Contract Programming).
+        )
+
+        $(P Preconditions and postconditions do not affect the type of the function.)
+
+    $(H3 Preconditions)
+
+        $(P An $(GLINK InContractExpression) is a precondition.)
+
+        $(P The first $(GLINK2 expression, AssignExpression) of the $(GLINK2 expression, AssertArguments)
+        must evaluate to true. If it does not, the precondition has failed.)
+
+        $(P The second $(I AssignExpression), if present, must be implicitly convertible to type `const(char)[]`.
+        )
 
-        $(P The $(D in) and $(D out) blocks or expressions of a function declaration specify
-        the pre- and post-conditions of the function. They are used in
-        $(LINK2 contracts.html, Contract Programming).
+        $(P An $(GLINK InStatement) is also a precondition. Any $(GLINK2 expression, AssertExpression) appearing
```

Comment:
boom `[UNDEFINED MACRO]         in an <!--UNDEFINED MACRO: "InStatement"--> will be an <i>InContractExpression</i>`